apps/api/src/domain/games/memory-match/repositories/arango-memory-match-repository.ts

```diff
@@ -86,6 +86,24 @@ export class ArangoMemoryMatchRepository implements IMemoryMatchRepository {
         }
     }
 
+    async unpublish(roundId: AggregateId): Promise<Error | AggregateId> {
+        try {
+            await this.database.update(roundId, { isPublished: false });
+            return roundId;
+        } catch (error) {
+            if (error?.message.includes('no document with that id')) {
+                return new InternalError(
+                    `You cannot publish memory match round: ${roundId}, as there is no memory match round with that ID.`
```

Comment:
Can we push this `try...catch` to the base arango implementation? The problem is that Arango's API depends on thrown errors and we treat many of these as returned (user input) errors.

Reply:
Next time we hit this, let's do the refactor.